marklogic-data-hub/src/main/java/com/marklogic/hub/dataservices/MappingService.java

```diff
@@ -0,0 +1,64 @@
+package com.marklogic.hub.dataservices;
+
+// IMPORTANT: Do not edit. This file is generated.
+
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+
+
+import com.marklogic.client.DatabaseClient;
+
+import com.marklogic.client.impl.BaseProxy;
+
+/**
+ * Provides a set of operations on the database server
+ */
+public interface MappingService {
+    /**
+     * Creates a MappingService object for executing operations on the database server.
+     *
+     * The DatabaseClientFactory class can create the DatabaseClient parameter. A single
+     * client object can be used for any number of requests and in multiple threads.
+     *
+     * @param db	provides a client for communicating with the database server
+     * @return	an object for session state
+     */
+    static MappingService on(DatabaseClient db) {
+        final class MappingServiceImpl implements MappingService {
+            private BaseProxy baseProxy;
+
+            private MappingServiceImpl(DatabaseClient dbClient) {
+                baseProxy = new BaseProxy(dbClient, "/data-hub/5/data-services/mapping/");
+            }
+
+            @Override
+            public com.fasterxml.jackson.databind.JsonNode testMapping(String docURI, String mappingName, String mappingVersion) {
+              return BaseProxy.JsonDocumentType.toJsonNode(
+                baseProxy
+                .request("testMapping.sjs", BaseProxy.ParameterValuesKind.MULTIPLE_ATOMICS)
+                .withSession()
+                .withParams(
+                    BaseProxy.atomicParam("docURI", false, BaseProxy.StringType.fromString(docURI)),
+                    BaseProxy.atomicParam("mappingName", false, BaseProxy.StringType.fromString(mappingName)),
+                    BaseProxy.atomicParam("mappingVersion", false, BaseProxy.StringType.fromString(mappingVersion)))
+                .withMethod("POST")
+                .responseSingle(false, Format.JSON)
```

Comment:
This is a GET method. Should the .api file be changed and the this file be regenerated ?

Reply:
I believe Data Services always generates  POST, I don't think there's a way to modify that - at least I don't see it in the docs. But does it matter? The client doesn't need to know.

Reply:
It doesn't matter, I will change the code to use POST method

Reply:
I'm not sure what you mean - the client to the interface doesn't care what HTTP method is used, and the Data Services endpoint doesn't care either. The Spring Boot controller can still be a "GET" request.

Reply:
Ok, I thought I had to change the Spring Boot controller. It works fine without any changes.